### PR TITLE
Improvements in HRID handling

### PR DESCRIPTION
**Pull request by fontanka16**
These four should all be done as of this PR:
https://github.com/FOLIO-FSE/folio_migration_tools/milestone/9

### CODE REVIEW

**Comment by fontanka16** [comment]
The lack of unit tests is a bit unfortunate, but I have validated the behaviour with the example data.

**Comment by bltravis** [comment]
@fontanka16 When you're preserving the 001 but here is a duplicate and you mint an HRID to use, instead, should the duplicate 001 (and 003, if it exists) be copied to an 035 (if that is enabled in the configuration, of course)?

File: src/folio_migration_tools/marc_rules_transformation/rules_mapper_bibs.py [hunk 1]
```diff
@@ -460,66 +461,76 @@ def handle_hrid(self, folio_instance, marc_record: Record, legacy_ids) -> None:
...
+        self.instance_hrid_counter += 1
+
+    def generate_numeric_part(self):
+        return (
+            str(self.instance_hrid_counter).zfill(11)
+            if self.common_retain_leading_zeroes
+            else str(self.instance_hrid_counter)
+        )
+
+    def preserve_001_as_hrid(self, folio_instance, marc_record, legacy_ids):
```
Comment (bltravis) [inline src/folio_migration_tools/marc_rules_transformation/rules_mapper_bibs.py:513]: Should we be copying the 001 + 003 when there is a duplicate in this case?

**Review by bltravis — COMMENTED** [review]
Re: previous comment

File: src/folio_migration_tools/marc_rules_transformation/rules_mapper_bibs.py (hunk 1 above)
Comment (fontanka16) [reply src/folio_migration_tools/marc_rules_transformation/rules_mapper_bibs.py:515]: Very good point. The idealistic part of me would like to have the library to fix this, but reality is that there will always be records that are not fixed. 

I made this change.

**Pull request merged**